src/states/client.rs

```diff
@@ -111,6 +111,13 @@ impl Client {
             Action::Terminate => {
                 return Transition::Terminate.to_evented();
             }
+            Action::ClientAuthority { result_tx } => {
+                let _ = result_tx.send(Authority::Client {
+                    client_key: *self.full_id.public_id().signing_public_key(),
+                    proxy_node_name: *self.proxy_public_id.name(),
+                    peer_id: self.crust_service.id(),
+                });
```

Comment:
Is it really necessary to add another `Action` just for this test?
Isn't it even possible to pass the client's key pair into its constructor? Then no change in the production code should be necessary at all.